### PR TITLE
Remove "ttl" from Cell and delete ExpiringKVS

### PR DESCRIPTION
**Pull request by gbonik**
This has never been used by anyone. If we actually wanted to support the
expiration dates in Cassandra, the Cell class would be a wrong place
to put the TTL because Cell represents a key, and TTL is not part of
a cell key. The fact that TTL is not part of the equals() and hashCode()
in that class is quite telling.

Not sure why Cell is marked as Serializable. I don't think any of our
products sends objects of this class over the network using Java
serialization or persists them in a database.

**Goals (and why)**:
Every time I look at the Cell class, the TTL stuff makes me very sad. I always wanted to remove it, and now I have some free time to do it (and someone reminded me of the issue).

**Concerns (what feedback would you like?)**:
Technically this breaks serialization, but I'm not sure why Cell was marked as Serializable in the first place.

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/2599)

### CODE REVIEW

**Comment by clockfort** [comment]
(source: I wrote this, and I agree that end-game we should delete this, as this ties transaction validity to wallclock times, which tickles me the wrong way)
A single internal product uses this to temporarily persist and re-serve high volume search results cache that sits in front of some extremely high latency search sources (1 day+ latency for the worst data sources); you would need to migrate off of them first.
You can do the migration very easily, if you test and promise them that the modern version of sweep that you've recently written can keep up with their throughput, on their 6-node C* cluster dedicated to this cache.

**Comment by gbonik** [comment]
That's interesting. I talked to Robert (one of the guys behind the mentioned product) and they indeed seem to be using it. However, they are on a two year old version of atlas and I'm not sure if they are ever planning to upgrade.

In the process of researching this I found out that we'd need to delete more stuff from TableMetadata and TableRenderer to complete remove this feature.

Will do more research; please don't merge this yet.

**Comment by gsheasby** [comment]
Ack, we'd like to see some of this cruft to go away, but are waiting for this research to be completed (hence the next action is on @gbonik).

**Comment by berler** [comment]
Considering that the only internal product using this feature is on a considerably old version of Atlas with no plans to upgrade, we probably shouldn't see them as a blocker as long as you have talked to them and they fully understand the steps they would need to take to migrate if they ever do decide to upgrade their version of Atlas.

**Comment by gbonik** [comment]
The consensus is that the mentioned internal product is not being actively developed or supported anymore (even though there is one production instance), and it doesn't make sense to block on it. The feature is very obscure and has absolutely zero test coverage.

I removed the corresponding bits of TableMetadata and code generation. I had to regenerate all schemas which makes the diff huge - please just ignore the generated files.

Which brings us to another question - why are we checking in the generated code. It is also impossible to regenerate the schema files without fixing compile errors in them first, which seems crazy. Basically, a chicken and egg problem - the `generateSchemas` gradle task depends on `compileJava`, which needs the schema files to be correct.

Removing "do not merge" and "WIP" labels now - this should be good to go if tests pass.

**Comment by gbonik** [comment]
Tests are broken on current develop if schemas are regenerated. I opened another PR to fix the issue: https://github.com/palantir/atlasdb

File: atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueServiceImpl.java [hunk 1]
```diff
@@ -1014,22 +1014,16 @@ protected int getMultiPutBatchCount() {
...
 
-    private void putInternal(final TableReference tableRef,
-                             final Iterable<Map.Entry<Cell, Value>> values) throws Exception {
-        putInternal(tableRef, values, CassandraConstants.NO_TTL);
-    }
-
-    protected void putInternal(final TableReference tableRef,
-                               Iterable<Map.Entry<Cell, Value>> values,
-                               final int ttl) throws Exception {
+    protected void putInternal(TableReference tableRef,
```
Comment (gsheasby) [inline atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueServiceImpl.java]: TODO(@gsheasby) check if this needs to be protected

File: docs/source/release_notes/release-notes.rst [hunk 2]
```diff
@@ -93,6 +93,10 @@ v0.64.0
            Previously, we would shut down the executors after a successful initialization, which could lead to a race condition with the submission of a ``cancelInitialization`` task.
            (`Pull Request <https://github.com/palantir/atlasdb/pull/2547>`__)
 
+    *    - |devbreak|
+         - Deleted the TTL duration field from the ``Cell`` class.
```
Comment (gsheasby) [inline docs/source/release_notes/release-notes.rst]: TODO (@gsheasby) list other dev-facing methods that have been removed here.

**Pull request merged**